### PR TITLE
Update Metadata docs

### PR DESCRIPTION
**Pull request by kuzdogan**

### CODE REVIEW

File: docs/metadata.rst [hunk 1]
```diff
@@ -157,7 +157,7 @@ to the end of the deployed bytecode
     0x00 0x33
 
 So in order to retrieve the data, the end of the deployed bytecode can be checked
-to match that pattern and use the IPFS hash to retrieve the file.
+to match that pattern and IPFS hash can be used to retrieve the file (if pinned/published).
```
Comment (leonardoalt) [inline docs/metadata.rst]:
```suggestion
to match that pattern and the IPFS hash can be used to retrieve the file (if pinned/published).
```

**Review by leonardoalt — COMMENTED** [review]
tiny nitpick

File: docs/metadata.rst [hunk 2]
```diff
@@ -181,14 +181,15 @@ Usage for Automatic Interface Generation and NatSpec
 ====================================================
 
 The metadata is used in the following way: A component that wants to interact
-with a contract (e.g. Mist or any wallet) retrieves the code of the contract,
-from that the IPFS/Swarm hash of a file which is then retrieved.  That file
+with a contract (e.g. Mist or any wallet) retrieves the code of the contract.
```
Comment (aarlt) [inline docs/metadata.rst]: Maybe we should not mention Mist anymore?

Comment (leonardoalt) [reply docs/metadata.rst]: yea I was also thinking the same

**Comment by kuzdogan** [comment]
Added detailed `userDoc` and `devDoc` descriptions.

File: docs/metadata.rst [hunk 3]
```diff
@@ -115,12 +115,66 @@ explanatory purposes.
...
+              // Contents of the @dev NatSpec field of the method
+              "details": "Returns a boolean value indicating whether the operation succeeded. Must be called by the token holder address",
+              // Contents of the @param NatSpec fields of the method
+              "params": {
+                "_value": "The amount tokens to be transferred",
+                "_to": "The receiver address"
+              }
+              // Contents of the @return NatSpec field.
+              "returns": {
+                // Return var name (here "success") if exist. "_0" as key if return var is unnamed
```
Comment (nikola-matic) [inline docs/metadata.rst]:
```suggestion
                // Return var name (here "success") if exists. "_0" as key if return var is unnamed
```

**Comment by nikola-matic** [comment]
FYI, `t_native_test_ext_gnosis ` failure was fixed, so you'll need to rebase.

**Comment by leonardoalt** [comment]
@kuzdogan the commits seem to be messed up after some merge/rebase. Can you please just rebase (not merge) your branch on top of `origin/develop`? Thanks!

**Comment by kuzdogan** [comment]
Oops yeah, should be clean now @leonardoalt

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**